pkg/storage/model_planStore.go

```diff
@@ -68,18 +68,19 @@ func (s *Store) ModelPlanGetByModelPlanIDLOADER(_ *zap.Logger, paramTableJSON st
 	return planSlice, nil
 }
 
-// ModelPlanCreate creates a model plan
-func (s *Store) ModelPlanCreate(logger *zap.Logger, plan *models.ModelPlan) (*models.ModelPlan, error) {
-
+// ModelPlanCreate creates a model plan using a transaction
+func (s *Store) ModelPlanCreate(np NamedPreparer, logger *zap.Logger, plan *models.ModelPlan) (*models.ModelPlan, error) {
 	if plan.ID == uuid.Nil {
 		plan.ID = uuid.New()
 	}
-	stmt, err := s.db.PrepareNamed(modelPlanCreateSQL)
+
+	stmt, err := np.PrepareNamed(modelPlanCreateSQL)
 	if err != nil {
 		logger.Error(
 			fmt.Sprintf("Failed to create model plan with error %s", err),
 			zap.String("user", plan.CreatedBy.String()),
 		)
+		// t.errors = append(t.errors, err) // TODO: SW should we just return the error? It gets appended in the parent transaction
```

Comment:
Good question, @ClayBenson94 what do you think? This has been a question before.

Reply:
sorry, this should be removed, just forgot to